doc/source/rllib/rllib-rlmodule.rst

```diff
@@ -0,0 +1,571 @@
+            """A RLModule with a shared encoder between agents for global observation."""
+
+            def __init__(
+                self, encoder: nn.Module, local_dim: int, hidden_dim: int, action_dim: int
+            ) -> None:
+                super().__init__(config=None)
+
+                self.encoder = encoder
+                self.policy_head = nn.Sequential(
+                    nn.Linear(hidden_dim + local_dim, hidden_dim),
+                    nn.ReLU(),
+                    nn.Linear(hidden_dim, action_dim),
+                )
+
+            def _forward_inference(self, batch):
+                with torch.no_grad():
+                    return self._common_forward(batch)
+
+            def _forward_exploration(self, batch):
+                with torch.no_grad():
+                    return self._common_forward(batch)
+            
+            def _forward_train(self, batch):
+                return self._common_forward(batch)
+
+            def _common_forward(self, batch):
+                obs = batch["obs"]
+                global_enc = self.encoder(obs["global"])
+                policy_in = torch.cat([global_enc, obs["local"]], dim=-1)
+                action_logits = self.policy_head(policy_in)
+
+                return {"action_dist": torch.distributions.Categorical(logits=action_logits)}
+
+
+
+        class BCTorchMultiAgentModuleWithSharedEncoder(MultiAgentRLModule):
+            def __init__(self, config: MultiAgentRLModuleConfig) -> None:
+                super().__init__(config)
+
+            def build(self):
+
+                module_specs = self.config.modules
+                module_spec = next(iter(module_specs.values()))
+                global_dim = module_spec.observation_space["global"].shape[0]
+                hidden_dim = module_spec.model_config_dict["fcnet_hiddens"][0]
+                shared_encoder = nn.Sequential(
+                    nn.Linear(global_dim, hidden_dim),
+                    nn.ReLU(),
+                    nn.Linear(hidden_dim, hidden_dim),
+                )
+
+                rl_modules = {}
+                for module_id, module_spec in module_specs.items():
+                    rl_modules[module_id] = BCTorchRLModuleWithSharedGlobalEncoder(
+                        encoder=shared_encoder,
+                        local_dim=module_spec.observation_space["local"].shape[0],
+                        hidden_dim=hidden_dim,
+                        action_dim=module_spec.action_space.n,
+                    )
+
+                self._rl_modules = rl_modules
+
+To construct this custom multi-agent RL odule, we need to pass the class to the :py:class:`~ray.rllib.core.rl_module.marl_module.MultiAgentRLModuleSpec` constructor. We need to also pass the :py:class:`~ray.rllib.core.rl_module.rl_module.SingleAgentRLModuleSpec` for each agent. This is because RLlib needs to know the observation, action spaces, and model hyper-parameters for each agent.
```

Comment:
```suggestion
To construct this custom multi-agent RL module, pass the class to the :py:class:`~ray.rllib.core.rl_module.marl_module.MultiAgentRLModuleSpec` constructor. Also, pass the :py:class:`~ray.rllib.core.rl_module.rl_module.SingleAgentRLModuleSpec` for each agent because RLlib requires the observation, action spaces, and model hyper-parameters for each agent.
```